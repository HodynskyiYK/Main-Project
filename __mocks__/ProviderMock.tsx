import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}