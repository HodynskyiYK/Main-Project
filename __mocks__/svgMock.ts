// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock