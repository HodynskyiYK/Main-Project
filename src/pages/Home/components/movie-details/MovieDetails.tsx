import React, { FC } from 'react'
import classnames from 'classnames'
import { Logo } from '../../../../components/logo'
import { Button, ButtonTypes } from '../../../../components/button'
import { SEARCH_ICON_29_30 } from '../../../../assets/svg'
import { concatMovieGenres, configureRuntime, getFullYear, getMovieImage } from '../../../../utils'
import { useMovieDetailsContext } from '../../../../context/movie-details-context'
import styles from './MovieDetails.module.scss'

export const MovieDetails: FC = () => {
    const {
        movieInfo,
        movieDetailsState,
        hideMovieDetails
    } = useMovieDetailsContext()

    if (!movieDetailsState || !movieInfo) {
        return null
    }

    return (
        <div className={classnames('container', styles.detailsContent)}>
            <div className={classnames('row', 'align-items-start', 'justify-content-between')}>
                <div className="col">
                    <Logo/>
                </div>
                <div className="col">
                    <Button
                        buttonClassName={ButtonTypes.ICON_BTN}
                        buttonText={SEARCH_ICON_29_30}
                        buttonAction={hideMovieDetails}
                    />
                </div>
            </div>
            <div className={classnames('row', 'align-items-start', styles.movieDetails)}>
                <div className="col col-4">
                    <div className={styles.movieImage}>
                        <img
                            src={movieInfo.poster_path}
                            alt={movieInfo.title}
                            onError={({ currentTarget }) => {
                                const defaultImage = getMovieImage('placeholder_320x480.png')
                                currentTarget.onerror = null // prevents looping
                                currentTarget.src=defaultImage
                            }}
                        />
                    </div>
                </div>
                <div className="col col-8">
                    <div className="row nowrap align-items-center">
                        <p className={classnames(styles.movieTitle, 'h1', 'col')}>{movieInfo.title}</p>
                        <div className="col">
                            <p className={classnames(styles.movieRating, 'col')}>
                                <span>{movieInfo.vote_average}</span>
                            </p>
                        </div>
                    </div>
                    <p className={styles.movieGenre}>{concatMovieGenres(movieInfo.genres)}</p>
                    <div className="row">
                        <div className="col">
                            <p className={styles.movieRelease}>{getFullYear(movieInfo.release_date)}</p>
                        </div>
                        <div className="col">
                            <p className={styles.movieDuration}>{configureRuntime(movieInfo.runtime)}</p>
                        </div>
                        <div className={classnames('col', 'col-12')}>
                            <p className={styles.movieDescription}>{movieInfo.overview}</p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
