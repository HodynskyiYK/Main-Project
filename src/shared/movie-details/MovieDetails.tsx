import React, { FC, useEffect } from 'react'
import classnames from 'classnames'
import queryString from 'query-string'
import { Logo } from '../../components/logo'
import { Button, ButtonTypes } from '../../components/button'
import { SEARCH_ICON_29_30 } from '../../assets/svg'
import { concatMovieGenres, configureRuntime, getFullYear, getMovieImage } from '../../utils'
import { useMovieDetailsContext } from '../../context/movie-details-context'
import styles from './MovieDetails.module.scss'
import { IMovieDetails } from './movieDetailsTypes'

export const MovieDetails: FC<IMovieDetails> = ({movieDetails}) => {
    const {
        movieInfo,
        movieDetailsState,
        hideMovieDetails
    } = useMovieDetailsContext()
    const {updateMovieDetails} = useMovieDetailsContext()

    useEffect(() => {
        if (movieDetails) {
            updateMovieDetails(movieDetails)
        }
    // eslint-disable-next-line
    }, [movieDetails])

    const backToSearAction = () => {
        const parsedParams = queryString.parse(location.search)
        if (parsedParams?.movie) {
            delete parsedParams.movie
        }
        hideMovieDetails()
    }

    if (!movieDetailsState || !movieInfo) {
        return null
    }

    return (
        <div className={classnames('container', styles.detailsContent)}>
            <div className={classnames('row', 'align-items-start', 'justify-content-between')}>
                <div className="col">
                    <Logo/>
                </div>
                <div className="col">
                    <Button
                        buttonClassName={ButtonTypes.ICON_BTN}
                        buttonText={SEARCH_ICON_29_30}
                        buttonAction={backToSearAction}
                    />
                </div>
            </div>
            <div className={classnames('row', 'align-items-start', styles.movieDetails)}>
                <div className="col col-4">
                    <div className={styles.movieImage}>
                        <img
                            src={movieInfo.poster_path}
                            alt={movieInfo.title}
                            onError={({ currentTarget }) => {
                                const defaultImage = getMovieImage('placeholder_320x480.png')
                                currentTarget.onerror = null // prevents looping
                                currentTarget.src=defaultImage
                            }}
                        />
                    </div>
                </div>
                <div className="col col-8">
                    <div className="row nowrap align-items-center">
                        <p className={classnames(styles.movieTitle, 'h1', 'col')}>{movieInfo.title}</p>
                        <div className="col">
                            <p className={classnames(styles.movieRating, 'col')}>
                                <span>{movieInfo.vote_average}</span>
                            </p>
                        </div>
                    </div>
                    <p className={styles.movieGenre}>{concatMovieGenres(movieInfo.genres)}</p>
                    <div className="row">
                        <div className="col">
                            <p className={styles.movieRelease}>{getFullYear(movieInfo.release_date)}</p>
                        </div>
                        <div className="col">
                            <p className={styles.movieDuration}>{configureRuntime(movieInfo.runtime)}</p>
                        </div>
                        <div className={classnames('col', 'col-12')}>
                            <p className={styles.movieDescription}>{movieInfo.overview}</p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
