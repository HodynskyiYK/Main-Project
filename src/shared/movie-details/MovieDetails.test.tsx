import React from 'react'
import { render } from '@testing-library/react'
import { MovieDetails } from './MovieDetails'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('Home tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock>
            <MovieDetails/>
        </ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})