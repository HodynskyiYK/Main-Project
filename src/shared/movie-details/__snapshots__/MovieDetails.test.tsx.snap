// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests To match snapshot 1`] = `<DocumentFragment />`;
